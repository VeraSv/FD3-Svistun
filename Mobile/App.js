"use strict";

import React from 'react';
import ReactDOM from 'react-dom';

import Mobile from './components/Mobile';
let companyName='Velcom';
let clients=[
    {name:'Иван', surname:'Иванов', middleName:'Иванович', balance:200},
    {name:'Сидор', surname:'Сидоров', middleName:'Сидорович', balance:250},
    {name:'Петр', surname:'Петров', middleName:'Петрович', balance:180},
    {name:'Григорий', surname:'Григорьев', middleName:'Григорьевич', balance:-220},
]
ReactDOM.render(<Mobile companyName={companyName} clients={clients}/>, document.getElementById('container') );

