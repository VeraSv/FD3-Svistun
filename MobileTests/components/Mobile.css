.Title{
  font-weight: 700;
  
}
.Table {
  border: solid 1px black;
  border-collapse: collapse;
 
}

td {
  border: solid 1px black;
  box-sizing: border-box;
  border-collapse: collapse;
}