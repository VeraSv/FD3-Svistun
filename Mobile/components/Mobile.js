import React from 'react';
import PropTypes from 'prop-types';
import MobileClient from './MobileClient';
import  './Mobile.css'
import {events} from './events';
import EditClient from './EditClient';
import AddClient from './AddClient';


class Mobile extends React.PureComponent {
  static propTypes = {
    companyName:PropTypes.string.isRequired,
    clients:PropTypes.arrayOf(
      PropTypes.shape({
        name: PropTypes.string.isRequired,
        surname: PropTypes.string.isRequired,
        middleName: PropTypes.string.isRequired,
        balance: PropTypes.number.isRequired,
      })
    ),
  }
  state = {
    companyName: this.props.companyName,
    clients: this.props.clients,
    list:[],
    edit:'',
    clientH:{},
    add:false  
  };

  setName1 = () => {
    this.setState({companyName:'Velcom'});
  };

  setName2 = () => {
    this.setState({companyName:'МТС'});
  };

allClients=() =>{
  this.setState({list:this.state.clients})
}

activeClients=()=>{
   let newClients=[...this.state.clients].filter(i=>i.balance>0)
   if(this.state.list.length!=newClients.length) {
  this.setState({list:newClients})
  }
}

blockedClients=()=>{
  let newClients=[...this.state.clients].filter(i=>i.balance<=0)
  if(this.state.list.length!=newClients.length) {
    this.setState({list:newClients})
    }
}

componentDidMount = () => {
 events.addListener('EditClicked',this.editClient);
 events.addListener('EditCard',this.editCard);
 events.addListener('Cancel',this.cancel);
 events.addListener('DeleteClicked',this.deleteClient);
 events.addListener('AddCard',this.addCard);
};

componentWillUnmount = () => {
  events.addListener('EditClicked',this.editClient);
  events.addListener('EditCard',this.editCard);
  events.addListener('Cancel',this.cancel);
  events.addListener('DeleteClicked',this.deleteClient);
  events.addListener('AddCard',this.addCard);
};

editClient=(client)=>{
this.setState({edit:true});
let editClient=this.state.clients.slice().filter(i=>i.name==client)
this.setState({clientH:editClient})
}

editCard=(newCard)=>{

    let changed=false;
    let newClients=[...this.state.clients]; 
    newClients.forEach( (c,i) => {
      if ( c.name==newCard.id) {
        newClients[i]=newCard
        changed=true;
      }
    } );
    if ( changed )
      this.setState({clients:newClients,list:newClients});
}

cancel=()=>{
  this.setState({edit:false, add:false});
}
  
deleteClient=(client)=>{
 
  let newClients=[...this.state.clients].filter(i=>i.name!=client)
  this.setState({clients:newClients, list:newClients})
}

addClient=()=>{
this.setState({add:true})
}
addCard=(value)=>{
  let newClients=[...this.state.clients,value]; 
 
    this.setState({clients:newClients,list:newClients})

}
  render() {
    console.log("MobileCompany render");
    var card=''
    if(this.state.edit==true) card=this.state.clientH.map(i=>
      { return <EditClient  key={i.name+1} id={i.name} name={i.name} surname={i.surname} middleName={i.middleName} balance={i.balance}
      />});
      if(this.state.add==true) card=<AddClient />
      if(!this.state.list.length) this.state.list=this.state.clients
var clients=this.state.list.map(i=>
  { return <MobileClient key={i.name} client={i}
  />})
    return (
      <div className='MobileCompany'>
        <input type="button" value="Velcom"  onClick={this.setName1}/>
        <input type="button" value="МТС"  onClick={this.setName2}/>
      
        <div className='MobileCompanyName'>Компания:{this.state.companyName}</div>
        <br></br>
        <input type="button" value="Все"  onClick={this.allClients}/>
        <input type="button" value="Активные"  onClick={this.activeClients}/>
        <input type="button" value="Заблокированные"  onClick={this.blockedClients}/>
        <table className='Table'>
          <tbody>
          <tr  key='title' className='Title'>
            <td> {"Фамилия"}</td>
           <td>{"Имя"}</td>
            <td>{"Отчество"}</td>
            <td>{"Баланс"} </td>
            <td>{"Статус"} </td>
            <td>{"Редактировать"} </td>
            <td>{"Удалить"} </td>
      
       </tr> 

         {clients}
            
       
      
          </tbody>
        </table>
        
        <input type="button" value="Добавить клиента"  onClick={this.addClient}/>
        <div>{card} </div>
      </div>
    )

  }

}

export default Mobile;
