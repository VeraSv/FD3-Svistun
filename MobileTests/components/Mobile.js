import React from 'react';
import PropTypes from 'prop-types';
import MobileClient from './MobileClient';
import  './Mobile.css'
import {events} from './events';
import EditClient from './EditClient';
import AddClient from './AddClient';


class Mobile extends React.PureComponent {
  static propTypes = {
    companyName:PropTypes.string,
    clients:PropTypes.arrayOf(
      PropTypes.shape({
        id:PropTypes.number.isRequired,
        name: PropTypes.string.isRequired,
        surname: PropTypes.string.isRequired,
        middleName: PropTypes.string.isRequired,
        balance: PropTypes.number.isRequired,
      })
    ),
  }
  state = {
    companyName: this.props.companyName,
    clients: this.props.clients,
    list:[],
    edit:'',
    clientH:{},
    add:false  
  };

 

allClients=() =>{
  this.setState({list:this.state.clients})
}

activeClients=()=>{
   let newClients=[...this.state.clients].filter(i=>i.balance>0)
   if(this.state.list.length!=newClients.length) {
  this.setState({list:newClients})
  }
}

blockedClients=()=>{
  let newClients=[...this.state.clients].filter(i=>i.balance<=0)
  if(this.state.list.length!=newClients.length) {
    this.setState({list:newClients})
    }
}

componentDidMount = () => {
 events.addListener('EditClicked',this.editClient);
 events.addListener('EditCard',this.editCard);
 events.addListener('Cancel',this.cancel);
 events.addListener('DeleteClicked',this.deleteClient);
 events.addListener('AddCard',this.addCard);
};

componentWillUnmount = () => {
  events.addListener('EditClicked',this.editClient);
  events.addListener('EditCard',this.editCard);
  events.addListener('Cancel',this.cancel);
  events.addListener('DeleteClicked',this.deleteClient);
  events.addListener('AddCard',this.addCard);
};

editClient=(client)=>{
this.setState({edit:true});
let editClient=this.state.clients.slice().filter(i=>i.id==client)
this.setState({clientH:editClient})
}

editCard=(newCard)=>{

    let changed=false;
    let newClients=[...this.state.clients]; 
    newClients.forEach( (c,i) => {
      if ( c.id==newCard.id) {
        newClients[i]=newCard
        changed=true;
      }
    } );
    if ( changed )
      this.setState({clients:newClients,list:newClients});
}

cancel=()=>{
  this.setState({edit:false, add:false});
}
  
deleteClient=(client)=>{
 
  let newClients=[...this.state.clients].filter(i=>i.id!=client)
  this.setState({clients:newClients, list:newClients})
}

addClient=()=>{
this.setState({add:true})
}
addCard=(value)=>{
  let newClients=[...this.state.clients,value]; 
 
    this.setState({clients:newClients,list:newClients})

}
  render() {
    console.log("MobileCompany render");
    var card=''
    if(this.state.edit==true) card=this.state.clientH.map(i=>
      { return <EditClient  key={i.id} id={i.id} name={i.name} surname={i.surname} middleName={i.middleName} balance={i.balance}
      />});
      if(this.state.add==true) {var x=Math.floor().random()*9+1; card=<AddClient id={x}/>}
      if(!this.state.list.length) this.state.list=this.state.clients
var clients=this.state.list.map(i=>
  { return <MobileClient key={i.id} client={i}
  />})
 
    return (
      <div className='MobileCompany'>
       
      
        <div className='MobileCompanyName'>Компания:{this.state.companyName}</div>
        <br></br>
        <input type="button" value="Все" id='All' onClick={this.allClients}/>
        <input type="button" value="Активные" id='Active' onClick={this.activeClients}/>
        <input type="button" value="Заблокированные" id='Blocked' onClick={this.blockedClients}/>
        <table className='Table'>
          <tbody>
          <tr  key='title' className='Title'>
            <td> {"Фамилия"}</td>
           <td>{"Имя"}</td>
            <td>{"Отчество"}</td>
            <td>{"Баланс"} </td>
            <td>{"Статус"} </td>
            <td>{"Редактировать"} </td>
            <td>{"Удалить"} </td>
      
       </tr> 

         {clients}
            
       
      
          </tbody>
        </table>
        
        <input type="button" value="Добавить клиента"  id='NewClient' onClick={this.addClient}/>
        <div>{card} </div>
      </div>
    )

  }

}

export default Mobile;
