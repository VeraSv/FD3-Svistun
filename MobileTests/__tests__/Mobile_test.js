"use strict";

import React from 'react';
import renderer from 'react-test-renderer';

import Mobile from '../components/Mobile.js';

test('работа Mobile', () => {
  let companyName='Velcom';
  let clients=[
    {id:1,name:'Иван', surname:'Иванов', middleName:'Иванович', balance:200},
    {id:2,name:'Сидор', surname:'Сидоров', middleName:'Сидорович', balance:250},
    {id:3,name:'Петр', surname:'Петров', middleName:'Петрович', balance:180},
    {id:4,name:'Григорий', surname:'Григорьев', middleName:'Григорьевич', balance:-220},
]
    const component = renderer.create(
        <Mobile companyName={companyName} clients={clients}/>
      );

      let componentTree=component.toJSON();
  expect(componentTree).toMatchSnapshot();

  const buttonElem = component.root.find((el) =>
  el.type == 'input' && el.props.className == 'All');
  buttonElem.props.onClick();

  const buttonElemActive = component.root.find((el) =>
  el.type == 'input' && el.props.className == 'Active');
  buttonElem.props.onClick();
  buttonElemActive.props.onClick();

  const buttonElemBlocked = component.root.find((el) =>
  el.type == 'input' && el.props.className == 'Blocked');
  buttonElem.props.onClick();
  buttonElemBlocked.props.onClick();

  componentTree=component.toJSON();
  expect(componentTree).toMatchSnapshot();

  buttonElem.props.onClick();
  buttonElemActive.props.onClick();
  buttonElemBlocked.props.onClick();


  componentTree=component.toJSON();
  expect(componentTree).toMatchSnapshot();

  const inst = component.getInstance();

  //Добавление
  let client1={id:15,name:'Сидор', surname:'Сидоров', middleName:'Сидорович', balance:50}
  inst.addCard(client1)
  expect(inst.state.clients.length).toEqual(5);

  //удаление
  
  inst.deleteClient(3);
    expect(inst.state.clients.length).toEqual(4);
   
  //редактирование
  
    let client={id:2,name:'Сидор', surname:'Сидоров', middleName:'Сидорович', balance:50}
    inst.editCard(client)
    expect(inst.state.clients[1].balance).toEqual(50);
  

  
  
})