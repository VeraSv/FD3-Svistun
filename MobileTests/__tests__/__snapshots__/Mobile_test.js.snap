// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`работа Mobile 1`] = `
<div
  className="MobileCompany"
>
  <div
    className="MobileCompanyName"
  >
    Компания:
    Velcom
  </div>
  <br />
  <input
    className="All"
    onClick={[Function]}
    type="button"
    value="Все"
  />
  <input
    className="Active"
    onClick={[Function]}
    type="button"
    value="Активные"
  />
  <input
    className="Blocked"
    onClick={[Function]}
    type="button"
    value="Заблокированные"
  />
  <table
    className="Table"
  >
    <tbody>
      <tr
        className="Title"
      >
        <td>
           
          Фамилия
        </td>
        <td>
          Имя
        </td>
        <td>
          Отчество
        </td>
        <td>
          Баланс
           
        </td>
        <td>
          Статус
           
        </td>
        <td>
          Редактировать
           
        </td>
        <td>
          Удалить
           
        </td>
      </tr>
      <tr>
        <td>
          Иванов
        </td>
        <td>
          Иван
        </td>
        <td>
          Иванович
        </td>
        <td>
          200
        </td>
        <td
          className="Active"
        >
          Active
        </td>
        <td>
          <input
            className="Edit"
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td>
           
          <input
            className="Delete"
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
           
        </td>
      </tr>
      <tr>
        <td>
          Сидоров
        </td>
        <td>
          Сидор
        </td>
        <td>
          Сидорович
        </td>
        <td>
          250
        </td>
        <td
          className="Active"
        >
          Active
        </td>
        <td>
          <input
            className="Edit"
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td>
           
          <input
            className="Delete"
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
           
        </td>
      </tr>
      <tr>
        <td>
          Петров
        </td>
        <td>
          Петр
        </td>
        <td>
          Петрович
        </td>
        <td>
          180
        </td>
        <td
          className="Active"
        >
          Active
        </td>
        <td>
          <input
            className="Edit"
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td>
           
          <input
            className="Delete"
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
           
        </td>
      </tr>
      <tr>
        <td>
          Григорьев
        </td>
        <td>
          Григорий
        </td>
        <td>
          Григорьевич
        </td>
        <td>
          -220
        </td>
        <td
          className="Blocked"
        >
          Blocked
        </td>
        <td>
          <input
            className="Edit"
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td>
           
          <input
            className="Delete"
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
           
        </td>
      </tr>
    </tbody>
  </table>
  <input
    className="NewClient"
    onClick={[Function]}
    type="button"
    value="Добавить клиента"
  />
  <div>
    
     
  </div>
</div>
`;

exports[`работа Mobile 2`] = `
<div
  className="MobileCompany"
>
  <div
    className="MobileCompanyName"
  >
    Компания:
    Velcom
  </div>
  <br />
  <input
    className="All"
    onClick={[Function]}
    type="button"
    value="Все"
  />
  <input
    className="Active"
    onClick={[Function]}
    type="button"
    value="Активные"
  />
  <input
    className="Blocked"
    onClick={[Function]}
    type="button"
    value="Заблокированные"
  />
  <table
    className="Table"
  >
    <tbody>
      <tr
        className="Title"
      >
        <td>
           
          Фамилия
        </td>
        <td>
          Имя
        </td>
        <td>
          Отчество
        </td>
        <td>
          Баланс
           
        </td>
        <td>
          Статус
           
        </td>
        <td>
          Редактировать
           
        </td>
        <td>
          Удалить
           
        </td>
      </tr>
      <tr>
        <td>
          Григорьев
        </td>
        <td>
          Григорий
        </td>
        <td>
          Григорьевич
        </td>
        <td>
          -220
        </td>
        <td
          className="Blocked"
        >
          Blocked
        </td>
        <td>
          <input
            className="Edit"
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td>
           
          <input
            className="Delete"
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
           
        </td>
      </tr>
    </tbody>
  </table>
  <input
    className="NewClient"
    onClick={[Function]}
    type="button"
    value="Добавить клиента"
  />
  <div>
    
     
  </div>
</div>
`;

exports[`работа Mobile 3`] = `
<div
  className="MobileCompany"
>
  <div
    className="MobileCompanyName"
  >
    Компания:
    Velcom
  </div>
  <br />
  <input
    className="All"
    onClick={[Function]}
    type="button"
    value="Все"
  />
  <input
    className="Active"
    onClick={[Function]}
    type="button"
    value="Активные"
  />
  <input
    className="Blocked"
    onClick={[Function]}
    type="button"
    value="Заблокированные"
  />
  <table
    className="Table"
  >
    <tbody>
      <tr
        className="Title"
      >
        <td>
           
          Фамилия
        </td>
        <td>
          Имя
        </td>
        <td>
          Отчество
        </td>
        <td>
          Баланс
           
        </td>
        <td>
          Статус
           
        </td>
        <td>
          Редактировать
           
        </td>
        <td>
          Удалить
           
        </td>
      </tr>
      <tr>
        <td>
          Григорьев
        </td>
        <td>
          Григорий
        </td>
        <td>
          Григорьевич
        </td>
        <td>
          -220
        </td>
        <td
          className="Blocked"
        >
          Blocked
        </td>
        <td>
          <input
            className="Edit"
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td>
           
          <input
            className="Delete"
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
           
        </td>
      </tr>
    </tbody>
  </table>
  <input
    className="NewClient"
    onClick={[Function]}
    type="button"
    value="Добавить клиента"
  />
  <div>
    
     
  </div>
</div>
`;
